Code Review #4

hoguer opened the pull request.

richardedwardbrown3 committed:
adding cheese and wine and cheese tbd

kbonifacio committed:
Merge branch 'dev' into felix/seedDataCreation

Felix-Cadiz committed:
fixed merge conflicts

Felix-Cadiz committed:
fixed merged conflicts

kbonifacio committed:
fixed backend connection to client. Server is running with no errors.

richardedwardbrown3 committed:
created folder tests, api and db spec js. Api has a test for products and the /products route. Db is coming soon.

richardedwardbrown3 committed:
made tests folder and api and db spec js

Felix-Cadiz committed:
feat(navbar) Navbar showing

kbonifacio committed:
completed semantic fixes across the db files.

michellemalf committed:
Merge branch 'dev' into felix/navigation

michellemalf committed:
Merge pull request #33 from hoguer/felix/navigation

Felix/navigation

michellemalf committed:
Merge branch 'dev' into kriselda/backendConnect

michellemalf committed:
Merge pull request #32 from hoguer/kriselda/backendConnect

Kriselda/backend connect

Felix-Cadiz committed:
feat(front end navigation), Navbar is connecting to different routes and endpoints.

Felix-Cadiz committed:
feat(components/products)

richardedwardbrown3 committed:
Merge pull request #35 from hoguer/felix/allProducts

feat (components/allProducts)

richardedwardbrown3 committed:
testing to see the tests git commit

richardedwardbrown3 committed:
fairly close to finished with the tests our code will respond to thus far.

michellemalf committed:
Michelle/frontend (#37)

* all products component not yet working

* products: not complete, need help

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

richardedwardbrown3 committed:
Met with Ryan, added to json test jest runInBand and test:watch as well as installed jsonWebToken.

richardedwardbrown3 committed:
Merge branch 'dev' into ricky/tests

richardedwardbrown3 committed:
Merge pull request #38 from hoguer/ricky/tests

Ricky/tests finally :)

Felix-Cadiz committed:
Felix/debug populate initial data (#42)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

Co-authored-by: Michelle M <89042600+michellemal@users.noreply.github.com>

kbonifacio committed:
Code Review #2 Fixes

kbonifacio committed:
Kriselda/products page (#47)

* added product api routes and router

* fixing merge conflicts to pull current dev branch

* code review 2 fixes #2

* feat(productPage): Displayed products and styled each card

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

Felix-Cadiz committed:
feat (routes/users) - User seed data created (#49)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

Felix-Cadiz committed:
Felix/fixed orders database adapters (#51)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

kbonifacio committed:
Kriselda/nav bar cart (#52)

* added product api routes and router

* fixing merge conflicts to pull current dev branch

* code review 2 fixes #2

* feat(productPage): Displayed products and styled each card

michellemalf committed:
latest changes to login and register, not yet working (#53)

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

richardedwardbrown3 committed:
This is the commit for my new branch with SingleProduct in it. (#50)

* This is the commit for my new branch with SingleProduct in it.

* couple changes

* Update init_db.js

* final commit tonight 1/31

* 5 fixes, env, ids, and

* now with all fixes and a pull from dev.

* Update db.spec.js

* Update db.spec.js

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

michellemalf committed:
Michelle/loginregister (#58)

* latest changes to login and register, not yet working

* more changes t oregister, not yet working

* register form

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

richardedwardbrown3 committed:
Fixing API and DB tests (#61)

kbonifacio committed:
Kriselda/single product client side (#55)

- Products showing with some CSS styling. 
- Users are able to access individual products and return to the all products page.

Felix-Cadiz committed:
Add orders routes and a middleware function to set the user on the req object (#59)

Commit:
Small Syntax Fixes (#63)

Commit:
fix merge conflicts

michellemalf committed:
Michelle/loginregister (#60)

Login and Register are both functional; however, the navigate feature after submission is not working.

michellemalf committed:
Michelle/loginregister (#66)

Login and Register Working!

michellemalf committed:
Michelle/account (#72)

Login, Register, and Account page is all functional.

kbonifacio committed:
Kriselda/update nav bar (#62)

Navbar is working and styled.

richardedwardbrown3 committed:
Ricky/products/productId test passes (#70)

Tests are functional.

kbonifacio committed:
Kriselda/products nav bar (#74)

Individual Wine and Cheese components have been updated.

Felix-Cadiz committed:
Felix/account styling  (#73)

account is functional, styling included

Commit:
Backend piece of Stripe setup (#71)

Starter code for Stripe component.

kbonifacio committed:
Kriselda/home component (#77)

Home component built and styled- functional.

kbonifacio committed:
Kriselda/products page2 (#84)

Styling and search params functioning!

Felix-Cadiz committed:
Felix/reviewsTable (#78)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* General styling to Login, Register and About Us.

* feat(reviews) Reviews db, seed data, and create function are functional.

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

michellemalf committed:
Michelle/adminbe (#85)

- Login and Register now working
- Styling done to the account page
- API Routes created

Commit:
Remove unused import

Felix-Cadiz committed:
Small Fixes for Code Review 3 (#69)

Commit:
Merge branch 'main' of github.com:hoguer/porto3000 into dev

Felix-Cadiz committed:
Felix/general styling - Home, Products, Login, and Register. (#90)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* comments addressed

* feat(searchbar) searchbar in products works for title, and description

* feat(products), Add to cart is still not functional

* feat(styling) general styling across home, products, login, and register.

* feat(account) general styling

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

kbonifacio committed:
Kriselda/winecheesepair1 (#89)

Code condensed, styling implemented, and seed data has been created.

Felix-Cadiz committed:
Felix/about page (#93)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* comments addressed

* feat(searchbar) searchbar in products works for title, and description

* feat(products), Add to cart is still not functional

* feat(addToCart) order is being made, but not linked to userId

* feat(addToCart) items can be added to the cart and linked to the userId

* fixed merge conflicts

* feat(about) images, text, and format update

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

Felix-Cadiz committed:
Felix/login register (#95)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* feat(login and Register) error messages now show on the front end

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

Felix-Cadiz committed:
Felix/order products frontend (#87)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* comments addressed

* feat(searchbar) searchbar in products works for title, and description

* feat(products), Add to cart is still not functional

* feat(addToCart) order is being made, but not linked to userId

* feat(addToCart) items can be added to the cart and linked to the userId

Co-authored-by: Kriselda Bonifacio <88959891+kbonifacio@users.noreply.github.com>

kbonifacio committed:
Kriselda/api payment routes (#91)

Added success and cancel routes through stripe.

michellemalf committed:
Michelle/initdb (#92)

* seed data for orders

richardedwardbrown3 committed:
Ricky/products/product (#94)

DB and API of order products

michellemalf committed:
Michelle/about (#96)

Updates to Michelle's about section.

michellemalf committed:
Michelle/about (#103)

all team members included in About; functional.

richardedwardbrown3 committed:
Ricky/products/product (#102)

Ricky Single Product

richardedwardbrown3 committed:
Ricky/products/product (#104)

SingleProduct

Commit:
Fix merge conflict changes that were not resolved

Commit:
Missing SingleOrder import in App component

hoguer commented on db/init_db.js:
```diff
@@ -459,6 +453,36 @@ async function createInitialUsers() {
+      {productId: "2", orderId: "2", price: "35", quantity: "2", userId:"2" },
+     ]
+
+    await Promise.all(orderProductsData.map(getOrdersByProduct));
```
I'll wait for @michellemal's PR before I merge this one, so that we have the correct function for creating initial order products.

kbonifacio replied:
updated and resolved.

hoguer commented on db/order_products.js:
```diff
@@ -0,0 +1,81 @@
+const client = require("./client");
+const { getProductById, getOrderById, updateOrder} = require("./")
+
+async function getOrderProductsById (id) {
```
this should be singular since we are just getting 1 `getOrderProductById`

kbonifacio replied:
updated and resolved.

hoguer commented on package.json:
```diff
@@ -10,7 +10,7 @@
     "db:build": "node ./db/init_db",
     "start": "node index.js",
-    "test": "jest --runInBand",
+    "test": "jest --runInBand --watchAll --verbose db",
```
we probably don't want to hardcode this to call `db` tests only.

kbonifacio replied:
updated and resolved.

hoguer commented on routes/index.js:
```diff
@@ -20,4 +22,8 @@ apiRouter.use((error, req, res, next) => {
+apiRouter.use((error, req, res, next) => {
+    res.send(error);
+});
+
```
It would be great if we could add a 404 (Page Not Found) handler above this one: 
```
// Create custom 404 handler that sets the status code to 404.
server.use('*', (req, res, next) =>{
  res.status(404);
  res.send({ error: 'route not found'});
})
```

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+const orderProductsRouter = require("express").Router();
+const { getOrderProductsById, addProductToOrder, updateOrderProduct, destroyOrderProduct} = require("../db")
+// const { isLoggedIn, isAdmin } = require("./util") need?
```
Since guests who are not logged in can create a cart, this can probably be removed from this file.

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orders.js:
```diff
@@ -1,4 +1,5 @@
 const ordersRouter = require("express").Router();
+const { user } = require("pg/lib/defaults");
```
we need to delete this line

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+});
+
+orderProductsRouter.patch("/:orderProductId", async (req, res, next) => {
+    const { orderId } = req.params;
```
this will need to be `const { orderProductId } = req.params;`

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+    const { orderId } = req.params;
+    const { productId, price, quantity } = req.body;
+    try {
+        const updatedOrderProducts = await updateOrderProduct({ orderId, productId, price, quantity })
```
Since you destructure the parameter as `id`, this has to be `updateOrderProduct({ id: orderProductId, productId, price, quantity })`

hoguer replied:
also the `const` should be singular: `updatedOrderProduct`

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+    const { orderId } = req.params;
+    const { productId, price, quantity } = req.body;
+    try {
+        const order_product = await addProductToOrder(orderId, productId, price, quantity)
```
`addProductToOrder` should take one parameter that is an object: `addProductToOrder({orderId, productId, price, quantity})`

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+    const { productId, price, quantity } = req.body;
+    try {
+        const order_product = await addProductToOrder(orderId, productId, price, quantity)
+        res.send({
```
this should be:
```
res.send({
  name: "Select item",
  message: "Your item is added to cart.",
  order_product
})
```

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+        res.send({
+            name: "Order Update",
+            message: "The order has been updated."
+        }, [updatedOrderProducts]);
```
this should be:
```
        res.send({
            name: "Order Product Update",
            message: "The order product has been updated.",
            updatedOrderProduct
        });
```

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+});
+
+orderProductsRouter.delete("/:orderProductId", async (req, res, next) => {
+    const { orderId } = req.params;
```
this should be `const { orderProductId } = req.params;`

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+
+orderProductsRouter.delete("/:orderProductId", async (req, res, next) => {
+    const { orderId } = req.params;
+    const { productId, price, quantity } = req.body;
```
we don't have a body for delete, this line should be removed.

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+    const { orderId } = req.params;
+    const { productId, price, quantity } = req.body;
+    try {
+        const destroyedOrderProduct = await destroyOrderProduct({orderId, productId, price, quantity})
```
this should be `await destroyOrderProduct(orderProductId)`

kbonifacio replied:
updated and resolved.

hoguer commented on routes/orderProducts.js:
```diff
@@ -0,0 +1,48 @@
+    const { productId, price, quantity } = req.body;
+    try {
+        const destroyedOrderProduct = await destroyOrderProduct({orderId, productId, price, quantity})
+        res.send({
```
this should be:
```
        res.send({
            name: "Order Product Deleted",
            message: "That order product is now removed.",
            destroyedOrderProduct
        });
```

kbonifacio replied:
updated and resolved.

hoguer commented on src/components/Login.js:
```diff
@@ -1,17 +1,18 @@
     const loginUser = () => {
 
-        if (!username && !password) {
+        if (!username || !password) {
```
looks good! This could even be on one line: `if (!username || !password) return;`

kbonifacio replied:
updated and resolved.

hoguer commented on src/components/Products.js:
```diff
@@ -22,41 +22,68 @@ const Products = ({products, setProducts}) => {
+        console.log("Add to Cart was pushed");
+        axios.post("/api/orders", {status, userId})
+            .then(res => { 
+                console.log("Adding item to order", res)
```
curious if we still need these logs?

kbonifacio replied:
will remove these in final product, still being used for debugging.

michellemalf committed:
Michelle/orderprodseed (#105)

Large update for orders_products.

Felix-Cadiz committed:
Felix/admin (#101)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* feat(login and Register) error messages now show on the front end

* fixed merge conflicts

* feat(adminAbilities) admin buttons are backend functional

* feat(adminPage) Not all components are functional

* feat(admin) Admin is able to add products

* feat(deleteProducts) Delete products is functional

* feat(admin) Styles update and feature layout

kbonifacio committed:
Kriselda/home comp update (#107)

Updated formatting and header. More code is present for final updates tonight.

kbonifacio committed:
Kriselda/code review4 (#108)

resolved conflicts from code review 4

Felix-Cadiz committed:
Felix/updates (#109)

* Tables working, servers connected.

* feat(allProducts), all products are not showing

* feat(allProducts) Tables are populating correctly

* fixed merge issues

* fixed merge conflict

* feat(routes/users)

* feat(ordersdb) fixed SQL.

* feat(orderRoutes), Orders API Routes are complete

* feat(routes), Checkout Databases and API Routes are complete

* Updates to API Routes & DB Adapters

* Code Review 3

* fixed merge conflicts

* feat(login and Register) error messages now show on the front end

* fixed merge conflicts

* feat(adminAbilities) admin buttons are backend functional

* feat(adminPage) Not all components are functional

* feat(admin) Admin is able to add products

* feat(deleteProducts) Delete products is functional

* feat(admin) Styles update and feature layout

* feat(order_products Orders show products

michellemalf committed:
Michelle/tests (#110)

* latest changes to login and register, not yet working

* more changes t oregister, not yet working

* register form

* register part 3

* Update Register.js

* register

* Remove log statement

* pass setCurrentUser in props

* pass setCurrentUser in props to Login, too

* register working, login is at 90%

* login register working

* Update .gitignore

Deleted extra text left from merge conflicts.

* account form working

* account working, week 4 backend, some account styling

* orders seed data

* resolved db/orders errors

* seed data for orders

* about me edits

* about edits

* initdb/orderproducts

* merge conflicts

* order products seed data

* more changes

* styled home

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>
Co-authored-by: Rachel Smith <rachel.smith@fullstackacademy.com>

kbonifacio committed:
Kriselda/product page css (#111)

product page styled and functional

michellemalf committed:
Michelle/review (#112)

Image alts created.

kbonifacio committed:
Kriselda/code clean up (#113)

code clean up

hoguer commented on db/init_db.js:
```diff
@@ -478,9 +502,29 @@ async function createInitialReviews() {
   }
 }
 
+async function addOrderProdcts() {
```
Is this a duplicate of the `createInitialReviews` function above?

hoguer replied:
while the name is different, it looks like this is still an exact copy of the function above, creating initial reviews.

hoguer commented on db/products.js:
```diff
@@ -69,10 +68,23 @@ async function patchProduct(id, fields = {}) {
     }
   }
 
+  async function destoryProduct({id}) {
```
should be `destroyProduct` (swap the o and the r in destory)

hoguer commented on db/products.js:
```diff
@@ -69,10 +68,23 @@ async function patchProduct(id, fields = {}) {
+  async function destoryProduct({id}) {
+    try {
+      const { rows: [product] } = await client.query(`
+      DELETE * FROM products
```
No `*` character for delete: `DELETE FROM products`

hoguer commented on db/products.js:
```diff
@@ -69,10 +68,23 @@ async function patchProduct(id, fields = {}) {
+      WHERE id=$1
+  `, [id]);
+    } catch (error) {
+      console.error("Error with deleteProduct in db/users.");
```
I think you want to change this to `db/products` instead of `db/users`

hoguer commented on db/products.js:
```diff
@@ -69,10 +68,23 @@ async function patchProduct(id, fields = {}) {
      getProductByName,
-     patchProduct
+     patchProduct,
+     destoryProduct
```
need to rename here, too: `destoryProduct` -> `destroyProduct`

hoguer commented on db/users.js:
```diff
@@ -100,12 +98,11 @@ async function patchUser(id, fields = {}) {
-      } = await client.query(`
-      DELETE FROM users
+      const { rows: [user] } = await client.query(`
+      DELETE * FROM users
```
This should be `DELETE FROM users` (no `*`)

hoguer commented on package.json:
```diff
@@ -21,11 +21,14 @@
     "bcrypt": "^5.0.1",
     "dotenv": "^15.0.0",
     "express": "^4.17.1",
+    "i": "^0.3.7",
```
curious what this package is and how we use it?

hoguer commented on package.json:
```diff
@@ -21,11 +21,14 @@
+    "i": "^0.3.7",
     "jsonwebtoken": "^8.5.1",
     "morgan": "^1.10.0",
+    "npm": "^8.5.0",
```
curious why we are adding `npm` itself in here?

hoguer commented on routes/index.js:
```diff
@@ -8,10 +8,21 @@ apiRouter.use("/users", usersRouter)
+  res.send({ error: 'route not found'});
+})
+
+apiRouter.use((error, req, res, next) => {
```
this is a duplicate of lines 26-28

hoguer commented on package.json:
```diff
@@ -21,11 +21,14 @@
     "pg": "^8.2.1",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-dropdown": "^1.9.2",
```
cool! Where do we use this?

hoguer commented on routes/payments.js:
```diff
@@ -2,25 +2,36 @@ require('dotenv').config();
-// TBD: modify to check isLoggedIn
-// Try it out: curl -X POST http://localhost:4000/api/payments/create-checkout-session
-paymentsRouter.post('/create-checkout-session', async (req, res) => {
+paymentsRouter.post('/create-checkout-session', isLoggedIn, async (req, res) => {
```
do we want to allow guests to checkout? (wondering about `isLoggedIn`)

hoguer replied:
Curious if this redirect is working after the user goes to the stripe checkout?
If not, we should have `const { DOMAIN_URL = 'http://porto3000.herokuapp.com/' } = process.env;`
And everyone should add this to their `.env`:
`DOMAIN_URL=http://localhost:3000`
Then update lines 17 and 18 to use `DOMAIN_URL` instead of `YOUR_DOMAIN`.

hoguer commented on routes/products.js:
```diff
@@ -59,5 +59,18 @@ productsRouter.patch('/:id', isAdmin, async (req, res, next)=>{
+productsRouter.delete('/:id', isAdmin, async (req, res, next) => {
+    try {
+        const { id } = req.params;
+        const destroyedProduct = await destoryProduct(req.id);
```
this should be `const destroyedProduct = await destoryProduct(id);` instead of `req.id`, since we get the `id` out of the params in the line above.

hoguer commented on routes/products.js:
```diff
@@ -59,5 +59,18 @@ productsRouter.patch('/:id', isAdmin, async (req, res, next)=>{
+      } catch (error) {
+        next({
+          name: "DeleteError",
+          message: "Could not delete user",
```
This message should be `Could not delete product`

hoguer commented on routes/users.js:
```diff
@@ -1,20 +1,19 @@
+    }
+} );
 
-// POST /api/users/register
```
Don't have to add them back, but these comments can actually be beneficial for immediately knowing the route to manually test a handler, instead of having to figure out where users.js is required, and then where that is required to determine what comes before `/register` if I want to try calling this thing from the client or via curl or postman, etc.

hoguer commented on routes/users.js:
```diff
@@ -135,7 +129,7 @@ usersRouter.patch('/:id', async (req, res, next)=>{
     try {
       const { id } = req.params;
-      const deletedUser = await deleteUser(id);
+      const deletedUser = await deleteUser(req.id);
```
curious why we changed this to `req.id` if we pull `id` out of `req.params` on the line above?

hoguer commented on src/components/About.js:
```diff
@@ -1,45 +1,62 @@
     return <>
-    <div className="aboutContainer">
-        <div className="about">
+    <div className="aboutHeader">
```
I love these descriptions :)

hoguer commented on src/components/Account.js:
```diff
@@ -1,63 +1,167 @@
   const [email, setEmail] = useState("");
   const [username, setUsername] = useState("");
   const [address, setAddress] = useState("");
+  const [isAdmin, setIsAdmin] = useState("");
```
We have this state both here and in App.js. It looks like we also have this information stored on the currentUser; is this necessary, or could we pass in currentUser to this component and call currentUser.isAdmin instead?

kbonifacio committed:
Kriselda/about component update (#115)

Updated CSS and cleaned up code: removed console.logs, unnecessary spaces, and added alts to images.

kbonifacio committed:
Krisleda/single product fixed (#117)

single product functional

richardedwardbrown3 committed:
Update products.js (#122)

fixed or to ro in "destroy" product

richardedwardbrown3 committed:
Update users.js (#126)

Addressed Rachel's comment.

Felix-Cadiz committed:
Felix/seed data stripe price (#120)

added stripe seed data

richardedwardbrown3 reviewed (commented):
Handled as much as I could

kbonifacio committed:
Kriselda/add to cart update (#127)

add to cart functional

michellemalf committed:
Update init_db.js (#129)

stripe price added again

richardedwardbrown3 committed:
finished cr4 (#128)

* finished cr4

* Update Account.js

Co-authored-by: Felix-Cadiz <88756231+Felix-Cadiz@users.noreply.github.com>

kbonifacio committed:
Kriselda/general css (#130)

Update styles & code cleanup

kbonifacio committed:
Kriselda/cart comp built (#133)

Ready for submission!

hoguer commented on db/init_db.js:
```diff
@@ -30,9 +32,11 @@ async function buildTables() {
           name VARCHAR(255) NOT NULL, 
           description VARCHAR(1000), 
           price INTEGER NOT NULL, 
+          stripe_price_id VARCHAR(255),
```
it looks like we accidentally added this twice

michellemalf replied:
removed

hoguer commented on index.js:
```diff
@@ -1,40 +1,35 @@
 server.use(express.static(path.join(__dirname, 'build')));
 
-// Authorization Middleware to jwt.verify and set req.user
+server.disable('etag')
```
this is good for testing in dev, but we should remove or comment this from our repo code.

hoguer commented on index.js:
```diff
@@ -1,40 +1,35 @@
       token = token.trim();
 
       const userObj = jwt.verify(token, JWT_SECRET);
+      console.log("userObj", userObj)
```
we should remove this log statement

hoguer commented on index.js:
```diff
@@ -1,40 +1,35 @@
 
-      // set the user on the request
       req.user = await getUserById(userObj.id);
+      console.log("req.user", req.user)
```
we should remove this log statement

Commit:
We want a 404 route handler AND an error handler

Commit:
Fix all react warnings when running client:dev (#136)

Commit:
Fix broken db:build (#135)

hoguer commented on db/orders.js:
```diff
@@ -1,3 +1,4 @@
+const paymentsRouter = require("../routes/payments");
```
we shouldn't be importing this here

michellemalf committed:
Michelle/review (#140)

michellemalf committed:
Update products.js (#137)

Commit:
Use Stripe (#141)

* fix conflicts

* update stripe to work

* navigate back to cart

* remove log

Commit:
Add ability to add multiple items to cart

Commit:
clean up package.json
